### PR TITLE
Fixing `grpc` dependencies and adding `abseil-cpp`.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

The `grpc` package suffered from unresolvable dependency issues:
1. It built its own version of `abseil-cpp` and `re2`, but wasn't packing any of the build libraries required during run-time.
2. Its dependency on `re2` couldn't be satisfied with the `re2` package we build separately.
3. The locally-build `abseil-cpp` libraries were excluded from the package, so they were never available and thus the package was failing to install.

###### Change Log  

- Added a new `abseil-cpp` package.
- Updated `cmake` and `gtest` to latest version to satisfy dependencies from `abseil-cpp`.
- Removed `*-static` subpackages from `gtest.spec`.
- Switched to using pre-built `re2` and `abseil-cpp` packages instead of building them separately for `grpc`.

###### Does this affect the toolchain?  

Yes.

###### Test Methodology

- Local package builds with `RUN_CHECK=y`.

### CODE REVIEW

**Review by jslobodzian — APPROVED** [review]

File: SPECS/abseil-cpp/abseil-cpp.spec [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Source0:        https://github.com/abseil/abseil-cpp/archive/%{version}/%{name}-%{version}.tar.gz
+# Workaround until GTest publishes a release including the "GTEST_FLAG_GET" macro.
+# Currently only available in the "main" branch: https://github.com/google/googletest/commit/977cffc4423a2d6c0df3fc9a7b5253b8f79c3f18
+Patch0:         gtest_build_fix.patch
+# Workaround until GTest publishes a release including the "::testing::Conditional" matcher.
+# Currently only available in the "main" branch: https://github.com/google/googletest/commit/8306020a3e9eceafec65508868d7ab5c63bb41f7
+Patch1:         disabling_invalid_tests.patch
+
+BuildRequires:  cmake >= 3.20.0
+BuildRequires:  gcc
```
Comment (anphel31) [inline SPECS/abseil-cpp/abseil-cpp.spec:18]: nit: these are always available in the build chroot, so it's not necessary to include them as BR

File: SPECS/grpc/grpc.spec [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
-Release:        6%{?dist}
+Release:        7%{?dist}
 License:        ASL 2.0
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/System
 URL:            https://www.grpc.io
-#Source0:        https://github.com/grpc/grpc/archive/v%{version}/%{name}-%{version}.tar.gz
-Source0:        %{name}-%{version}.tar.gz
+Source0:        https://github.com/grpc/grpc/archive/v%{version}/%{name}-%{version}.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (anphel31) [inline SPECS/grpc/grpc.spec:10]: is the comment at the end of this line necessary?

**Review by anphel31 — APPROVED** [review]
LGTM. would recommend a toolchain build for validation

File: SPECS/grpc/grpc.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/grpc/grpc.spec:10]: Oh, right, it's not.:) Thanks!

File: SPECS/abseil-cpp/abseil-cpp.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/abseil-cpp/abseil-cpp.spec:18]: There's no harm in keeping them there and being explicit about build dependencies seems like a good idea, IMO. Although I'm aware we never really capture all of the most basic ones. And I think `cmake` is part of the toolchain but not part of the chroot.

**Review by PawelWMS — COMMENTED** [review]
A full package rebuild revealed issues while building `mariadb`, `libiothsm-std`, and `azure-iotedge`. All of them seem to be related to the new version of `cmake`.

**Pull request merged**